src/fitnesse/socketservice/SocketService.java

```diff
@@ -71,7 +71,7 @@ private void serviceThread() {
       } catch (SocketException sox) {
         running = false; // do nothing
       } catch (IOException e) {
-        throw new RuntimeException(e);
+        LOG.log(Level.SEVERE, "I/O exception in service thread", e);
```

Comment:
This should not be propagated? This seems contrary to the purpose of the pull request?
Maybe a comment why this may be logged but not propagated is in place.